### PR TITLE
[Release Tooling] Copy over macOS/macCatalyst plists

### PR DESCRIPTION
**Pull request by ncooke3**

### CODE REVIEW

**Review by paulb777 — APPROVED** [review]
LGTM on green and verification of https://github.com/firebase/firebase-ios-sdk/issues/12505#issuecomment-1988972581

File: ReleaseTooling/Sources/ZipBuilder/FrameworkBuilder.swift [hunk 1]
```diff
@@ -674,7 +674,17 @@ struct FrameworkBuilder {
       let binaryName = frameworkPath.lastPathComponent.replacingOccurrences(of: ".framework",
                                                                             with: "")
       let fatBinary = frameworkPath.appendingPathComponent(binaryName).resolvingSymlinksInPath()
-      let infoPlist = frameworkPath.appendingPathComponent("Info.plist").resolvingSymlinksInPath()
+      let plistPathComponents = {
+        if platform == .catalyst || platform == .macOS {
```
Comment (paulb777) [inline ReleaseTooling/Sources/ZipBuilder/FrameworkBuilder.swift:678]: Run style after verifying the zip builds

Comment (ncooke3) [reply ReleaseTooling/Sources/ZipBuilder/FrameworkBuilder.swift:678]: yep, have that and changelog done locally. will push now that zip is done. Still working on verifying when using all artifacts from the zip. Encountering some 401 network errors at the moment.

**Comment by ncooke3** [comment]
Verified app depending on entire zip (minus GSI, Admob) using https://github.com/firebase/firebase-ios-sdk/issues/12505#issuecomment-1988972581 method.

Because the Analytics frameworks are now out-of-date, I had to manually add `Info.plist` to them (simulating this PR) to get the verification to work.

**Pull request merged**